apps/app/src/views/routes/Prescriptions.tsx

```diff
@@ -193,26 +213,46 @@ export const Prescriptions = () => {
   const [finished, setFinished] = useState<boolean>(false);
   const [filterTextDebounce] = useDebounce(filterText, 250);
 
-  const { prescriptions, loading, error, refetch } = getPrescriptions({
-    patientId,
-    prescriberId,
-    patientName: filterTextDebounce.length > 0 ? filterTextDebounce : null
-  });
+  const getPrescriptionsData = {
+    patientId: patientId || undefined,
+    prescriberId: prescriberId || undefined,
+    state: status,
+    patientName: filterTextDebounce.length > 0 ? filterTextDebounce : undefined
+  };
+
+  const { prescriptions, loading, error, refetch } = getPrescriptions(getPrescriptionsData);
 
   useEffect(() => {
     if (!loading) {
       const preppedRows = prescriptions.filter((row) => row).map(renderRow);
       setRows(preppedRows);
       setFinished(prescriptions.length === 0);
     }
-  }, [loading]);
+  }, [loading, prescriptions]);
+
+  useEffect(() => {
+    navigate({
+      search: status ? `?status=${status}` : ''
+    });
+    async function refetchData() {
+      setRows([]);
+      setFilterChangeLoading(true);
+      const { data } = await refetch(getPrescriptionsData);
+      setFilterChangeLoading(false);
```

Comment:
Shouldnt `loading` on line 223 get set to true once we refetch?

Reply:
I thought so at first but it's not... but probably for the best. If loading were set to `true` on refetch then the whole list would switch to skeleton rows rather than the spinner for infinite scroll as it's written.